feat(err): add fingerprint embeddings table

oliverb123 opened the pull request.

oliverb123 committed:
first pass

github-actions[bot] committed:
Update query snapshots

oliverb123 committed:
Merge branch 'master' into err/add-fingerprint-embeddings-table

oliverb123 committed:
fix topic name

fuziontech committed:
fix(clickhouse): Update fingerprint embeddings migration to use proper node roles and ingestion layer

Updated migration 0153 to follow the modern ingestion layer pattern:
- Data table (ReplacingMergeTree) on DATA nodes
- Writable distributed table on INGESTION_SMALL nodes
- Kafka table and materialized view on INGESTION_SMALL nodes
- All ingestion layer components use on_cluster=False

This aligns with the pattern established in migrations 0151/0152 and ensures tables are created on the correct node types in the ClickHouse topology.

github-actions[bot] committed:
Update query snapshots

Daesgar commented on posthog/models/error_tracking/sql.py:
```diff
@@ -81,3 +86,97 @@
+        table_name=KAFKA_ERROR_TRACKING_FINGERPRINT_EMBEDDINGS_TABLE,
+        on_cluster_clause=ON_CLUSTER_CLAUSE(on_cluster),
+        engine=kafka_engine(
+            KAFKA_ERROR_TRACKING_ISSUE_FINGERPRINT_EMBEDDINGS, group="clickhouse-error-tracking-fingerprint-embeddings"
```
This is a silly request, but if this is a new table, could we use `_` instead? We have a chaotic mix of group names, and we'll probably soon refactor those so they are all `clickhouse_whatever` which will allow us to ease their monitoring. 🙏 

```suggestion
            KAFKA_ERROR_TRACKING_ISSUE_FINGERPRINT_EMBEDDINGS, group="clickhouse_error_tracking_fingerprint_embeddings"
```

fuziontech committed:
Apply suggestion from @Daesgar

Co-authored-by: Daniel Escribano <daniel.e@posthog.com>

fuziontech committed:
refactor: Remove on_cluster parameters from fingerprint embeddings SQL functions

The embeddings tables should never run on cluster, so remove the on_cluster parameter to prevent misuse. All functions now hardcode on_cluster=False.

github-actions[bot] committed:
Update query snapshots

oliverb123 committed:
Update posthog/models/error_tracking/sql.py

Co-authored-by: Daniel Escribano <daniel.e@posthog.com>

github-actions[bot] committed:
Update query snapshots

fuziontech committed:
Merge branch 'master' into err/add-fingerprint-embeddings-table
